docs/af5-getting-started/modules/ROOT/pages/implement-stateful-event-handler.adoc

```diff
@@ -0,0 +1,433 @@
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.axonframework.modelling.StateManager;
+import org.axonframework.modelling.annotation.InjectEntity;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class WhenAllCoursesFullyBookedThenSendNotification {
+
+    private static final String FACULTY_ID = "ONLY_FACULTY_ID"; // <1>
+
+    @EventSourcedEntity // <2>
+    record State(Map<CourseId, Course> courses, boolean notified) {
+
+        record Course(int capacity, int students) { // <3>
+
+            Course capacity(int newCapacity) {
+                return new Course(newCapacity, this.students);
+            }
+
+            Course studentSubscribed() {
+                return new Course(this.capacity, this.students + 1);
+            }
+
+            Course studentUnsubscribed() {
+                return new Course(this.capacity, this.students - 1);
+            }
+
+            boolean isFullyBooked() { // <4>
+                return students >= capacity;
+            }
+        }
+
+        @EntityCreator // <5>
+        State() {
+            this(new HashMap<>(), false);
+        }
+
+        @EventSourcingHandler // <6>
+        State evolve(CourseCreated event) {
+            courses.put(event.courseId(), new Course(event.capacity(), 0));
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(CourseCapacityChanged event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.capacity(event.capacity()));
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(StudentSubscribedToCourse event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentSubscribed());
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(StudentUnsubscribedFromCourse event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentUnsubscribed());
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(AllCoursesFullyBookedNotificationSent event) { // <7>
+            return new State(courses, true);
+        }
+    }
+    
+    // Command and Event handlers will be added in next steps...
+}
+----
+
+<1> Simplified to use a single faculty ID for this tutorial
+<2> Event-sourced entity that maintains state across multiple events
+<3> Inner record representing individual course state with capacity and current enrollment
+<4> Business logic to determine if a course is fully booked
+<5> Default constructor creating empty state
+<6> Event sourcing handlers that evolve state based on different event types
+<7> Handler to track when notification has already been sent - thanks to that you can track it in your EventStore instead of external systems
```

Comment:
Minor nit for readability:
```suggestion
<7> Handler to track when notification has already been sent - allows you to track it in your EventStore instead of external systems
```